library/templates/pharmacy_list_form.php

```diff
@@ -0,0 +1,437 @@
+<?php
+
+/**
+ * Handles the creating and updating of the weno alternate and primary pharmacies. 
+ * Saving of these properties are handled by the scripts
+ * that call the LBF form.  The current example can be found in 
+ * interface/patient_file/summary/demographics_save.php
+ * and in interface/new/new_comprehensive_save.php
+ *
+ * @package   OpenEMR
+ * @link      https://www.open-emr.org
+ *
+ * @author    Kofi Appiah <kkappiah@medsov.com>
+ * @copyright Copyright (c) 2023 Omega Systems Group International. <info@omegasystemsgroup.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+require_once $GLOBALS['srcdir'] . '/options.inc.php';
+
+use OpenEMR\Modules\WenoModule\Services\PharmacyService;
+use OpenEMR\Common\Csrf\CsrfUtils;
+use OpenEMR\Common\Acl\AclMain;
+use OpenEMR\Common\Twig\TwigContainer;
+
+if(!AclMain::aclCheckCore('patients', 'med')){
+    echo (new TwigContainer(null, $GLOBALS['kernel']))->getTwig()->render('core/unauthorized.html.twig', ['pageTitle' => xl("Pharmacy Selector")]);
+    exit;
+}
+
+$widgetConstants = [
+    'listWithAddButton' => 26
+    ,'textDate' => 4
+    ,'textbox' => 2
+];
+
+global $pid; // we need to grab our pid from our global settings.
```

Comment:
above line should never need to be done. $pid will always be set to current patient that is selected (if a patient has been selected)

Reply:
actually, disregard above note. you are calling this from within a function, so ok to do above line.